services/QuillLMS/engines/comprehension/app/models/comprehension/rule.rb

```diff
@@ -68,14 +68,21 @@ def determine_feedback_from_history(feedback_history)
 
     def regex_is_passing?(entry)
       regex_rules.none? do |regex_rule|
-        regex_rule.sequence_type == RegexRule::TYPE_INCORRECT ? Regexp.new(regex_rule.regex_text).match(entry) : !Regexp.new(regex_rule.regex_text).match(entry)
+        regex_match = match_with_case_sensitive(regex_rule.regex_text, regex_rule.case_sensitive, entry)
+        # for "incorrect" type regex rules, we want to "fail" if they have the regex. for "required" type regex
+        # rules, we want to "fail" when they dont have the regex.
+        regex_rule.sequence_type == RegexRule::TYPE_INCORRECT ? regex_match : !regex_match
       end
     end
 
     def display_name
       DISPLAY_NAMES[rule_type.to_sym] || rule_type
     end
 
+    private def match_with_case_sensitive(regex, is_case_sensitive=false, text)
```

Comment:
It almost feels like this function should be on the `RegexRule` model.  You could avoid passing in both `regex` and `is_case_sensitive` here and just do something like `RegexRule.check_for_match(text)`.  It's probably not a big deal, but I think that getting to avoid passing this data around is probably a plus.